### PR TITLE
Document PR documentation requirements & numpydoc format switch

### LINKED ISSUE #314: Question: should we move to numpydoc format?
Many libraries use the [`numpydoc`](https://numpydoc.readthedocs.io/en/latest/format.html) format, which is understood by many tools in the ecosystem as well.

Would there be any interest in moving to it instead of what we have?

### PR DESCRIPTION
**Pull request by romanlutz**
Closes #314 raised by @adrinjalali 

Signed-off-by: Roman Lutz <rolutz@microsoft.com>

### CODE REVIEW

File: docs/contributor_guide/development_process.rst [hunk 1]
```diff
@@ -14,6 +14,18 @@ run periodically. When adding new code paths or features, tests are a
 requirement to complete a pull request. They should be added in the
 :code:`test` directory.
 
+Documentation should be provided with pull requests that add or change
```
Comment (MiroDudik) [inline docs/contributor_guide/development_process.rst:17]: Do we want to mention CHANGES.md and AUTHORS.md

File: docs/contributor_guide/development_process.rst [hunk 2]
```diff
@@ -14,6 +14,18 @@ run periodically. When adding new code paths or features, tests are a
...
+functionality. This includes comments in the code itself, docstrings, and user
+guides. For exceptions to this rule the pull request author should coordinate
+with a maintainer.
+
+Docstrings should follow
+`numpydoc format <https://numpydoc.readthedocs.io/en/latest/format.html>`_.
+This is a
+`recent decision by the community <https://numpydoc.readthedocs.io/en/latest/format.html>`_.
```
Comment (MiroDudik) [inline docs/contributor_guide/development_process.rst:25]: bad URL

File: docs/contributor_guide/development_process.rst (hunk 1 above)
Comment (romanlutz) [reply docs/contributor_guide/development_process.rst:17]: Good point

File: docs/contributor_guide/development_process.rst [hunk 3]
```diff
@@ -14,6 +14,22 @@ run periodically. When adding new code paths or features, tests are a
...
+with a maintainer. For changes that fix bugs, add new features, change APIs,
+etc., i.e., for changes that are relevant to developers and/or users please
+also add an entry in CHANGES.md in the section corresponding to the *next*
+release, since that's where your change will be included.
+If you're a new contributor please also add yourself to AUTHORS.md.
+
+Docstrings should follow
+`numpydoc format <https://numpydoc.readthedocs.io/en/latest/format.html>`_.
+This is a
+`recent decision by the community <https://github.com/fairlearn/fairlearn/issues/314>`_.
```
Comment (riedgar-ms) [inline docs/contributor_guide/development_process.rst:29]: Shouldn't this be linking to something in the proposals repo?

Comment (romanlutz) [reply docs/contributor_guide/development_process.rst:29]: Does it matter? We can move it there if it's important to you, but at the end of the day it's just a closed issue.

**Pull request merged**